### PR TITLE
change gas reference from u256 to uint64_t in every exec method"

### PR DESCRIPTION
**Pull request by winsvega**
DO NOT MERGE

### CODE REVIEW

File: test/libevm/vm.cpp [hunk 1]
```diff
@@ -184,6 +184,7 @@ void FakeExtVM::importExec(mObject& _o)
 	gasPrice = toInt(_o["gasPrice"]);
 	gas = toInt(_o["gas"]);
 	execGas = gas;
+	ugas = uint64_t(gas);
```
Comment (winsvega) [inline test/libevm/vm.cpp:187]: use an existing gas variable.

File: test/libevm/vm.cpp [hunk 2]
```diff
@@ -334,7 +335,7 @@ void doVMTests(json_spirit::mValue& _v, bool _fillin)
 			auto vmtrace = Options::get().vmtrace ? fev.simpleTrace() : OnOpFunc{};
 			{
 				Listener::ExecTimeGuard guard{i.first};
-				output = vm->exec(fev.gas, fev, vmtrace);
+				output = vm->exec(fev.ugas, fev, vmtrace);
```
Comment (winsvega) [inline test/libevm/vm.cpp:338]: use an exsiting gas variable

File: libethereum/Executive.cpp [hunk 3]
```diff
@@ -375,13 +377,14 @@ bool Executive::go(OnOpFunc const& _onOp)
...
+				uint64_t gas = uint64_t(m_gas);
 				if (m_res)
 				{
-					m_res->output = vm->exec(m_gas, *m_ext, _onOp); // take full output
+					m_res->output = vm->exec(gas, *m_ext, _onOp); // take full output
 					bytesConstRef{&m_res->output}.copyTo(m_outRef);
 				}
 				else
-					vm->exec(m_gas, *m_ext, m_outRef, _onOp); // take only expected output
+					vm->exec(gas, *m_ext, m_outRef, _onOp); // take only expected output
```
Comment (winsvega) [inline libethereum/Executive.cpp:387]: exec accept gas by reference don't use temporary variables

**Comment by chfast** [comment]
Wanted to do that! Thanks.

I'm using `int64_t` (signed) in evmjit as the limit is 2^63-1.

**Comment by gcolvin** [comment]
Wanted to do what where @chfast ?

**Comment by gcolvin** [comment]
@winsvega @chfast  I am closing this and backing out these changes for now.  I've got a clean compile, but most any test that counts gas fails, and there are too many interrelated diffs to know where to start.  Plus, the new JIT interface uses int64_t, whereas the new VM uses uint64_t.  I've more important changes to make, and we can get back to this more systematically.

**Pull request closed without merging**